components/graphs/Graph.tsx

```diff
@@ -19,13 +19,29 @@ import {
 import useTime from '~/lib/hooks/useTime';
 import { ClearSkyDataPoint, ForecastDataPoint } from '~/lib/types';
 
-function getGraphStartDate(currentTime: number) {
+function getGraphStartDate(currentTime: number, totalHours: number) {
   const currentDate = new Date(currentTime);
   return new Date(
     currentDate.getFullYear(),
     currentDate.getMonth(),
     currentDate.getDate(),
-    currentDate.getHours() - 5
+    totalHours > 1
+      ? currentDate.getHours() - totalHours / 8
+      : currentDate.getHours(),
+    totalHours > 1 ? 0 : currentDate.getMinutes() - 15
+  );
+}
+
+function getGraphEndDate(currentTime: number, totalHours: number) {
+  const currentDate = new Date(currentTime);
+  return new Date(
+    currentDate.getFullYear(),
+    currentDate.getMonth(),
+    currentDate.getDate(),
+    totalHours > 1
+      ? currentDate.getHours() + (7 * totalHours) / 8
+      : currentDate.getHours(),
+    totalHours > 1 ? 0 : currentDate.getMinutes() + 45
```

Comment:
same here, everything's working great but would like description of behaviour in a comment for maintainability